### ISSUE TITLE
[Bug]: mo panic when run queries from specific view by stack overflow

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#7810]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):d54992c96efce308c17d37a638fb147a8fce6863
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

mysql> drop table if exists t;
Query OK, 0 rows affected (0.05 sec)

mysql> drop VIEW if exists v;
Query OK, 0 rows affected (0.07 sec)

mysql> CREATE TABLE t(f1 INTEGER);
Query OK, 0 rows affected (0.18 sec)

mysql> insert into t values(1),(2),(3),(6);
Query OK, 4 rows affected (0.08 sec)

mysql> CREATE VIEW v AS SELECT f1 FROM t;
Query OK, 0 rows affected (0.10 sec)

mysql> SELECT f1 FROM (SELECT f1 FROM v) AS dt1 NATURAL JOIN v dt2 WHERE f1 > 5;
+------+
| f1   |
+------+
|    6 |
+------+
1 row in set (0.01 sec)

mysql> 
mysql> SELECT f1 FROM v NATURAL JOIN v dt2 WHERE f1 > 5;
+------+
| f1   |
+------+
|    6 |
+------+
1 row in set (0.05 sec)

mysql> 
mysql> 
mysql> ALTER VIEW v AS SELECT f1 FROM (SELECT f1 FROM v) AS dt1 NATURAL JOIN v dt2 WHERE f1 > 5;
Query OK, 0 rows affected (0.05 sec)

mysql> 
mysql> 
mysql> SELECT * FROM v;

ERROR 2013 (HY000): Lost connection to MySQL server during query
No connection. Trying to reconnect...
ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1:6001' (61)
ERROR: 
Can't connect to the server

**Comment by ouyuanning** [comment]
get it. i will try to fix it tomorrow.

**Comment by daviszhen** [comment]
I think it was fixed at #7840

**Comment by aressu1985** [comment]
fixed: 4410d8618193455fc0be25d3af17d19b5bbc1a0e


mysql> use test;
Database changed
mysql> drop table if exists t;
Query OK, 0 rows affected (0.02 sec)

mysql> drop VIEW if exists v;
Query OK, 0 rows affected (0.02 sec)

mysql> CREATE TABLE t(f1 INTEGER);
Query OK, 0 rows affected (0.06 sec)

mysql> insert into t values(1),(2),(3),(6);
Query OK, 4 rows affected (0.03 sec)

mysql> CREATE VIEW v AS SELECT f1 FROM t;
Query OK, 0 rows affected (0.07 sec)

mysql>  SELECT f1 FROM (SELECT f1 FROM v) AS dt1 NATURAL JOIN v dt2 WHERE f1 > 5;
+------+
| f1   |
+------+
|    6 |
+------+
1 row in set (0.02 sec)

mysql>  SELECT f1 FROM v NATURAL JOIN v dt2 WHERE f1 > 5;
+------+
| f1   |
+------+
|    6 |
+------+
1 row in set (0.00 sec)

mysql> ALTER VIEW v AS SELECT f1 FROM (SELECT f1 FROM v) AS dt1 NATURAL JOIN v dt2 WHERE f1 > 5;
ERROR 20101 (HY000): internal error: there is a recursive reference to the view v
mysql>
